packages/core/src/routes/session.test.ts

```diff
@@ -5,38 +5,320 @@ import { createRequester } from '@/utils/test-utils';
+        expect.stringContaining('/session/consent')
+      );
+    });
+
+    it('throw error with other prompt name', async () => {
+      getInteractionDetails().mockResolvedValue({
+        prompt: { name: 'invalid' },
+      });
+      await expect(sessionRequest.post('/session').send({})).resolves.toHaveProperty('status', 400);
+    });
+  });
+
+  describe('POST /session/sign-in/username-password', () => {
+    it('assign result and redirect', async () => {
+      const response = await sessionRequest.post('/session/sign-in/username-password').send({
+        username: 'username',
+        password: 'password',
+      });
+      expect(response.statusCode).toEqual(200);
+      expect(response.body).toHaveProperty('redirectTo');
+      expect(getProvider().interactionResult).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.anything(),
+        expect.objectContaining({ login: { accountId: 'user1' } }),
+        expect.anything()
+      );
+    });
+
+    it('throw if user not found', async () => {
+      const response = await sessionRequest.post('/session/sign-in/username-password').send({
+        username: 'notexistuser',
+        password: 'password',
+      });
+      expect(response.statusCode).toEqual(500);
+    });
+  });
+
+  describe('POST /session/sign-in/passwordless/phone/send-passcode', () => {
+    beforeAll(() => {
+      getInteractionDetails().mockResolvedValue({
+        jti: 'jti',
+      });
+    });
+    it('call sendPasscode', async () => {
+      const response = await sessionRequest
+        .post('/session/sign-in/passwordless/phone/send-passcode')
+        .send({ phone: '13000000000' });
+      expect(response.statusCode).toEqual(204);
+      expect(sendPasscode).toHaveBeenCalled();
+    });
+    it('throw error if phone not exists', async () => {
+      const response = await sessionRequest
+        .post('/session/sign-in/passwordless/phone/send-passcode')
+        .send({ phone: '13000000001' });
+      expect(response.statusCode).toEqual(422);
+    });
+  });
+
+  describe('POST /session/sign-in/passwordless/phone/verify-passcode', () => {
+    it('assign result and redirect', async () => {
+      const response = await sessionRequest
+        .post('/session/sign-in/passwordless/phone/verify-passcode')
+        .send({ phone: '13000000000', code: '1234' });
+      expect(response.statusCode).toEqual(200);
+      expect(response.body).toHaveProperty('redirectTo');
+      expect(getProvider().interactionResult).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.anything(),
+        expect.objectContaining({ login: { accountId: 'user1' } }),
+        expect.anything()
+      );
+    });
+    it('throw error if verifyPasscode failed', async () => {
+      const response = await sessionRequest
+        .post('/session/sign-in/passwordless/phone/verify-passcode')
+        .send({ phone: '13000000000', code: '1231' });
+      expect(response.statusCode).toEqual(500);
+    });
+  });
```

Comment:
nit: not sure whether should we add a test case of phone number not existing in this route. or test of a non-exist phone number test case in the send-passcode route is good enough?